File: ruby-quest04-dev/ruby-quest04/ex00/my_string_index.rb
# The function my_string_index takes two parameters: haystack and needle.

def my_string_index(haystack, needle)
    index = 0
  
    while index < haystack.length
      if haystack[index] == needle
        return index
      end
      index += 1
    end
  
    return -1  # Return -1 if no match is found
  end

  #puts my_string_index("Hello", "l")
  #puts my_string_index("aaaaa", "b")
  #puts my_string_index("Hello World", "w")
  #puts my_string_index("Hello World", "W")

File: js-quest06-dev/js-quest06/ex00/my_string_index.js
/*

#1 What are the inputs?

#2 What are the output?

#3 What is the function doing?

*/
function my_string_index(haystack, needle) {
    var index = 0;

    while(index < haystack.length) {
        if(haystack[index] == needle) {
            return index;
        }
        index += 1;
    }

    return -1;

}
